### ISSUE TITLE
"lib/project requires a projectRoot"

### ISSUE THREAD

**Issue by blodgettza** [cypress-io/cypress#756]
- Operating System: Windows 10
- Cypress Version: 1.0.2
- Browser Version: 61

### Is this a Feature or Bug?
Error

### Current behavior:
When I try and run  `cypress run --record --key xxxxxx-xxxxxx-xxxxxx-xxxxxx` I get the error: `Error: Instantiating lib/project requires a projectRoot!`

### Desired behavior:
I would like to record the test so I can view the results in the browser Dashboard

### How to reproduce:
Follow the installation process when downloading the zip file and try to cypress run.

#### Test code:

```js

```

**Comment by masquadel** [comment]
I ran into the same Error.
It might have to do with this:
[this](https://docs.cypress.io/dashboard/overview/projects-dashboard.html#Authentication)

Is it necessary to put it up into a Github project for it to run?
¯\\_(ツ)_/¯

**Comment by bahmutov** [comment]
@blodgettza can you list what you have done to recreate the bug
- create new project with `npm init -y`
- add cypress with `npm i -D cypress`
- execute `cypress run --record ...` 
Did you open cypress once first so it loads and scaffolds the `cypress/...` folder?

**Comment by jgauffin** [comment]
I downloaded the zip file from: https://docs.cypress.io/guides/getting-started/installing-cypress.html#Direct-Download

I then followed the instructions here: https://docs.cypress.io/guides/core-concepts/dashboard-service.html

It's when I try to run cypress at step 10 that I get the error.

Do I need to run cypress from a certain folder? I tried both in the root of the project and in the generated cypress folder.

Cypress itself is in "C:\apps\cypress" and my project is in "C:\src\myproject\src"

**Comment by richdom81** [comment]
I too have this same issue.

I created a project at dashboard.cypress.io and I have the ProjectID and KEY. 

I ran the 'cypress run --record - -key xxxxx'  and get the error above.
I've tried running cypress from the .exe location and the local project location I created 'c:\CypressTests'
I've also added the -P flag to point to the local project location and I still get the same error.

**Comment by jgauffin** [comment]
ping @bahmutov

**Comment by bahmutov** [comment]
Hmm, I have not investigated this on Windows. And you have used `--run-project` flag?

**Comment by kutlaykural** [comment]
Hi! @bahmutov 
I tried as you say; 
cypress --run-project --record --key xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

But now it says;
TypeError: s.indexOf is not a function

![adsiz](https://user-images.githubusercontent.com/38225108/38696038-af07be76-3e96-11e8-8574-05d887bee044.png)

**Comment by jennifer-shehane** [comment]
@brian-mann Yes, this would fall under this issue in the documentation https://github.com/cypress-io/cypress-documentation/issues/427

**Comment by brian-mann** [comment]
@bahmutov and @jennifer-shehane we may also want to warn the user when they attempt to invoke the binary directly with arguments as opposed to going through the CLI.

We can detect that by passing a `--run-from-cli` argument to the binary, and when absent we know it's been invoked incorrectly.

**Comment by bahmutov** [comment]
Yes, we should do this detection and warning
